src/app/activity/bounty/Publish.ts

```diff
@@ -49,15 +59,15 @@ const getDbHandler = async (bountyId: string, guildId: string, request: PublishR
 	const dbCollectionBounties = db.collection('bounties');
 	const dbCollectionCustomers = db.collection('customers');
 
-	const status = request.clientSyncRequest ? 'Open' : 'Draft';
+	let search = { _id: new mongo.ObjectId(bountyId) };
+
+	// If we are syncing from web, status should be Open
+	if (request.clientSyncRequest) search["status"] = 'Open';
 
 	const dbCustomerResult: CustomerCollection = await dbCollectionCustomers.findOne({
 		customerId: guildId
 	});
-	const dbBountyResult: BountyCollection = await dbCollectionBounties.findOne({
-		_id: new mongo.ObjectId(bountyId),
-		status: status,
-	});
+	const dbBountyResult: BountyCollection = await dbCollectionBounties.findOne(search);
```

Comment:
If my understanding is correct, nothing has changed about the functionality, though you are using the full bounty record to search for a bounty. Just for my understanding when I look back at this in the future, why so?

Reply:
It's just searching on the ID unless we are doing a client sync, when we need to limit the search to just Open bounties to make sure we don't have a race condition.